mailchimp_auth/mailchimp.py

```diff
@@ -0,0 +1,185 @@
+from django.conf import settings
+import email_normalize
+
+import mailchimp_marketing as MailchimpMarketing
+from mailchimp_marketing.api_client import ApiClientError
+
+
+class MailchimpAPI(object):
+    '''
+    Wrapper for supporter methods:
+    https://mailchimp.com/developer/marketing/api/list-members/
+    '''
+    LIST_ID = settings.MAILCHIMP_LIST_ID
+    API_KEY = settings.MAILCHIMP_API_KEY
+    SERVER = settings.MAILCHIMP_SERVER
+
+    def _has_valid_email(self, supporter, email_address):
+        '''
+        Determine whether a supporter has a valid contact matching the given
+        email address.
+        '''
+        email_valid = (email_normalize.normalize(supporter["email_address"]) == email_normalize.normalize(email_address) and
+                        supporter['status'] != 'unsubscribed')
+
+        if email_valid:
+            return True
+
+        return False
+
+    def put_supporter(self, user):
+        '''
+        Add or update supporter.
+        '''
+
+        payload = {
+            "email_address": user.email,
+            "status": "subscribed",
+            "merge_fields": {
+                "FNAME": user.first_name,
+                "LNAME": user.last_name,
+            }
+        }
+
+        subscriber = payload["email_address"]
+
+        try:
+            client = MailchimpMarketing.Client()
```

Comment:
hm.. what about setting up the client once in an __init__ method for this class?